src/bulletin/bulletin.service.ts

```diff
@@ -69,4 +69,98 @@ export class BulletinService {
       }
     });
   }
+
+  async updateBulletin(
+    id: number,
+    updateBulletinDto: BulletinDTO,
+    pdfAttachments: Express.Multer.File[],
+  ) {
+    return this.prismaService.$transaction(async (tx) => {
+      try {
+        const updatedBulletin = await tx.bulletin.update({
+          where: { id },
+          data: {
+            category_id: updateBulletinDto.category_id,
+            title: updateBulletinDto.title,
+            content: updateBulletinDto.content,
+            author: updateBulletinDto.author,
+          },
+        });
+
+        if (pdfAttachments && pdfAttachments.length > 0) {
+          // This remove old records
+          const oldAttachments = await tx.pDFAttachment.findMany({
+            where: { bulletin_id: id, deleted_at: null },
+          });
+
+          for (const attachment of oldAttachments) {
+            const references = await tx.bulletin.findMany({
+              where: {
+                pdfAttachments: { some: { id: attachment.id } },
+                deleted_at: null,
+              },
+            });
+
+            if (references.length === 0) {
+              const { error } = await this.supabaseService
+                .getSupabase()
+                .storage.from(process.env.STORAGE_BUCKET)
+                .remove([attachment.file_path]);
+
+              if (error) {
+                throw new Error(`Failed to remove old file: ${error.message}`);
+              }
+            }
+          }
+
+          await tx.pDFAttachment.updateMany({
+            where: { bulletin_id: id },
+            data: { deleted_at: new Date() },
+          });
+
+          // This create new records
+          for (const pdfAttachment of pdfAttachments) {
+            const now = new Date();
+            const formattedDate = `${String(now.getDate()).padStart(
+              2,
+              '0',
+            )}-${String(now.getMonth() + 1).padStart(
+              2,
+              '0',
+            )}-${now.getFullYear()}`;
+            const sanitizedFileName = pdfAttachment.originalname
+              .replace(/\s+/g, '-')
+              .replace(/[^a-zA-Z0-9.-]/g, '');
+            const uniqueFileName = `${formattedDate}-${sanitizedFileName}`;
+            const filePath = uniqueFileName;
```

Comment:
Does it have to be in this format? Can't we use date-fns for formatting? @SaaammmyyyS 
If we really have a specified format, we can do a custom util function pero if this is doing normal formatting lang, let's just utilize date-fns